Unable to build and run tests on architectures without unsafe SIMD support

# Before you file an issue

- Did you read the docs? *yes*
- Did you read the README? *yes*

# The problem

Working on any architecture other than `x86_64` or `aarch64`, such as `i686`, `ppc64le`, or `s390x`:

```
$ git clone https://github.com/marshallpierce/rust-base64.git
$ cd rust-base64
$ cargo build
[…]
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 10.00s
$ cargo test
[…]
error[E0432]: unresolved import `crate::engine::simd`
  --> src/engine/tests.rs:6:20
   |
 6 | use crate::engine::simd::Simd;
   |                    ^^^^ could not find `simd` in `engine`
   |
note: found an item that was configured out
  --> src/engine/mod.rs:24:9
   |
19 |       any(
   |  ________-
20 | |         target_arch = "x86_64",
21 | |         all(target_arch = "aarch64", target_feature = "neon")
22 | |     )
   | |_____- the item is gated here
23 |   ))]
24 |   pub mod simd;
   |           ^^^^

```

(For the above, I used an `qemu-user-static`-emulated Fedora Rawhide `ppc64le` [`mock` chroot](https://github.com/rpm-software-management/mock). I first observed this issue while working on updating Fedora’s [`rust-base64` package](https://src.fedoraproject.org/rpms/rust-base64).)

# How I, the issue filer, am going to help solve it

- I will open a PR that improves conditionalization in `src/engine/tests.rs`.

--- a/src/engine/tests.rs
+++ b/src/engine/tests.rs
@@ -2,7 +2,13 @@
 #![allow(unused_variables)]
 
 use crate::alphabet::Symbol;
-#[cfg(feature = "simd-unsafe")]
+#[cfg(all(
+    feature = "simd-unsafe",
+    any(
+        target_arch = "x86_64",
+        all(target_arch = "aarch64", target_feature = "neon")
+    )
+))]
 use crate::engine::simd::Simd;
 #[cfg(all(
     feature = "simd-unsafe",
@@ -43,7 +49,16 @@ use std::{collections, fmt, io::Read as _};
 #[case::general_purpose(GeneralPurposeWrapper)]
 #[case::naive(NaiveWrapper)]
 #[case::decoder_reader(DecoderReaderEngineWrapper)]
-#[cfg_attr(feature = "simd-unsafe", case::simd(SimdEngineWrapper))]
+#[cfg_attr(
+    all(
+        feature = "simd-unsafe",
+        any(
+            target_arch = "x86_64",
+            all(target_arch = "aarch64", target_feature = "neon")
+        )
+    ),
+    case::simd(SimdEngineWrapper)
+)]
 #[cfg_attr(
     all(
         feature = "simd-unsafe",
@@ -72,8 +87,26 @@ fn engines_supporting_arbitrary_alphabets<E: EngineWrapper>(#[case] engine_wrapp
 #[template]
 #[rstest]
 // so that there's at least one engine with no features enabled
-#[cfg_attr(not(feature = "simd-unsafe"), case::naive(NaiveWrapper))]
-#[cfg_attr(feature = "simd-unsafe", case::simd(SimdEngineWrapper))]
+#[cfg_attr(
+    not(all(
+        feature = "simd-unsafe",
+        any(
+            target_arch = "x86_64",
+            all(target_arch = "aarch64", target_feature = "neon")
+        )
+    )),
+    case::naive(NaiveWrapper)
+)]
+#[cfg_attr(
+    all(
+        feature = "simd-unsafe",
+        any(
+            target_arch = "x86_64",
+            all(target_arch = "aarch64", target_feature = "neon")
+        )
+    ),
+    case::simd(SimdEngineWrapper)
+)]
 #[cfg_attr(
     all(
         feature = "simd-unsafe",
@@ -105,7 +138,16 @@ enum CommonAlphabet {
 #[rstest]
 #[case::general_purpose(GeneralPurposeWrapper)]
 #[case::naive(NaiveWrapper)]
-#[cfg_attr(feature = "simd-unsafe", case::simd(SimdEngineWrapper))]
+#[cfg_attr(
+    all(
+        feature = "simd-unsafe",
+        any(
+            target_arch = "x86_64",
+            all(target_arch = "aarch64", target_feature = "neon")
+        )
+    ),
+    case::simd(SimdEngineWrapper)
+)]
 #[cfg_attr(
     all(
         feature = "simd-unsafe",
@@ -1912,10 +1954,22 @@ impl EngineWrapper for DecoderReaderEngineWrapper {
     }
 }
 
-#[cfg(feature = "simd-unsafe")]
+#[cfg(all(
+    feature = "simd-unsafe",
+    any(
+        target_arch = "x86_64",
+        all(target_arch = "aarch64", target_feature = "neon")
+    )
+))]
 struct SimdEngineWrapper;
 
-#[cfg(feature = "simd-unsafe")]
+#[cfg(all(
+    feature = "simd-unsafe",
+    any(
+        target_arch = "x86_64",
+        all(target_arch = "aarch64", target_feature = "neon")
+    )
+))]
 impl EngineWrapper for SimdEngineWrapper {
     type Engine = Simd;
 

